### PR TITLE
langchain: add document_variable_name to create_stuff_documents_chain

### PR DESCRIPTION
**Pull request by xdumaine**
- **Description:** `StuffDocumentsChain` uses `LLMChain` which is deprecated by langchain runnables. `create_stuff_documents_chain` is the replacement, but needs support for `document_variable_name` to allow multiple uses of the chain within a longer chain.
- **Issue:** none
- **Dependencies:** none

### CODE REVIEW

File: libs/langchain/langchain/chains/combine_documents/stuff.py [hunk 1]
```diff
@@ -91,9 +96,16 @@ def format_docs(inputs: dict) -> str:
     ).with_config(run_name="stuff_documents_chain")
 
 
+@deprecated(
```
Comment (eyurtsev) [inline libs/langchain/langchain/chains/combine_documents/stuff.py:99]: @xdumaine we can merge the rest of this except for the deprecation. We're trying to figure out how to deprecate some of these things without causing too much churn for users that have existing code that's relying on this. 

cc @ccurme  -- has the most knowledge here in terms of which chains are safe to deprecate

**Review by eyurtsev — CHANGES_REQUESTED** [review]
We can merge after the `@deprecation` is removed. We're mostly want the @ deprecated decorator to be used by core maintainers since we have to plan the roll out carefully for each deprecation

File: libs/langchain/langchain/chains/combine_documents/stuff.py (hunk 1 above)
Comment (xdumaine) [reply libs/langchain/langchain/chains/combine_documents/stuff.py]: @eyurtsev no problem, thank you for the review/feedback!

**Review by eyurtsev — APPROVED** [review]

**Comment by xdumaine** [comment]
Thank you!

**Pull request merged**